src/GitHub.App/Models/RepositoryHost.cs

```diff
@@ -1,168 +1,39 @@
 ﻿using System;
 using System.Diagnostics;
-using System.Diagnostics.CodeAnalysis;
-using System.Globalization;
-using System.Linq;
-using System.Reactive;
-using System.Reactive.Linq;
-using System.Reactive.Threading.Tasks;
-using System.Threading.Tasks;
 using GitHub.Api;
-using GitHub.Authentication;
-using GitHub.Caches;
 using GitHub.Extensions;
 using GitHub.Primitives;
 using GitHub.Services;
-using NLog;
-using Octokit;
 using ReactiveUI;
+using static System.FormattableString;
 
 namespace GitHub.Models
 {
     [DebuggerDisplay("{DebuggerDisplay,nq}")]
     public class RepositoryHost : ReactiveObject, IRepositoryHost
     {
-        static readonly Logger log = LogManager.GetCurrentClassLogger();
-
-        readonly ILoginManager loginManager;
-        readonly HostAddress hostAddress;
-        readonly IKeychain keychain;
-        readonly IUsageTracker usage;
-
-        bool isLoggedIn;
+        readonly IConnection connection;
 
         public RepositoryHost(
+            IConnection connection,
             IApiClient apiClient,
-            IModelService modelService,
-            ILoginManager loginManager,
-            IKeychain keychain,
-            IUsageTracker usage)
+            IModelService modelService)
         {
+            Guard.ArgumentNotNull(connection, nameof(connection));
+            Guard.ArgumentNotNull(apiClient, nameof(apiClient));
+            Guard.ArgumentNotNull(modelService, nameof(modelService));
+
+            this.connection = connection;
             ApiClient = apiClient;
             ModelService = modelService;
-            this.loginManager = loginManager;
-            this.keychain = keychain;
-            this.usage = usage;
-
-            Debug.Assert(apiClient.HostAddress != null, "HostAddress of an api client shouldn't be null");
-            Address = apiClient.HostAddress;
-            hostAddress = apiClient.HostAddress;
-            Title = apiClient.HostAddress.Title;
-        }
-
-        public HostAddress Address { get; private set; }
-
-        public IApiClient ApiClient { get; private set; }
-
-        public bool IsLoggedIn
-        {
-            get { return isLoggedIn; }
-            private set { this.RaiseAndSetIfChanged(ref isLoggedIn, value); }
         }
 
-        public string Title { get; private set; }
-
-        [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope")]
-        public IObservable<AuthenticationResult> LogInFromCache()
-        {
-            Func<Task<AuthenticationResult>> f = async () =>
-            {
-                try
-                {
-                    var user = await loginManager.LoginFromCache(Address, ApiClient.GitHubClient);
-                    var accountCacheItem = new AccountCacheItem(user);
-                    usage.IncrementLoginCount().Forget();
```

Comment:
TODO: Need to work out if a login from cache counts as a login, and if so put this in `ConnectionManager.

Reply:
Done this now in `ConnectionManager`.